### PR TITLE
op-challenger: fix the wrong variable name and some additional fix

### PR DESCRIPTION
**Pull request by DeeADouble**
[fix variable name](https://github.com/ethereum-optimism/optimism/commit/470ebfb0e627f7f7e248881fe9f70a70ad967361)

[fix from differential-testing](https://github.com/ethereum-optimism/optimism/commit/d73b9ccc553514952b07ada6ea92e2a841819dec)

[fix from L2Genesis](https://github.com/ethereum-optimism/optimism/commit/614b67be4b5a508e65e3b61e7a231e3655b29552)

### CODE REVIEW

File: op-challenger/game/fault/contracts/metrics/metrics.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-const ConstractSubsystem = "contracts"
+const ConstructSubsystem = "contracts"
```
Comment (Inphi) [inline op-challenger/game/fault/contracts/metrics/metrics.go]: Should be Contracts not Construct.

Comment (DeeADouble) [reply op-challenger/game/fault/contracts/metrics/metrics.go]: Yeah, you are right thanks

**Comment by DeeADouble** [comment]
@Inphi PLZ have a check, thanks

**Review by Inphi — APPROVED** [review]
Thanks

**Pull request merged**